homeassistant/components/switchbot/config_flow.py

```diff
@@ -151,6 +158,57 @@ async def async_step_password(
             },
         )
 
+    async def async_step_lock_auth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the SwitchBot API auth step."""
+        errors = {}
+        assert self._discovered_adv is not None
+        if user_input is not None:
+            try:
+                key_details = await self.hass.async_add_executor_job(
+                    SwitchbotLock.retrieve_encryption_key,
+                    self._discovered_adv.address,
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                )
+                return await self.async_step_lock_key(key_details)
+            except RuntimeError:
```

Comment:
Is it not possible to catch a more narrow exception?

Reply:
I looked into the library, and besides the broad RuntimeError that is raised, we're not catching potential errors of a `requests.post` call.

Reply:
I guess I could add a custom exception in the underlying library. What do you think @bdraco , is it worth the trouble of releasing a new version?

Reply:
Happy to do another release for this 👍

Reply:
https://github.com/home-assistant/core/pull/84888